pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1507,24 +1556,42 @@ public CompletableFuture<Void> close(
         CompletableFuture<Void> closeFuture = new CompletableFuture<>();
 
         lock.writeLock().lock();
+        CompletableFuture<Void> inProgressTransferCloseTask = null;
         try {
             if (!disconnectClients) {
                 transferring = true;
             }
             // closing managed-ledger waits until all producers/consumers/replicators get closed. Sometimes, broker
             // forcefully wants to close managed-ledger without waiting all resources to be closed.
-            if (!isClosingOrDeleting || closeWithoutWaitingClientDisconnect) {
-                fenceTopicToCloseOrDelete();
+            if (isClosingOrDeleting) {
+                if (!disconnectClients) {
```

Comment:
nit: add a comment.

// Dedup the first-phase close from Transfer. Do not run the first-phase close multiple times.